syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.leveldb";
option java_outer_classname = "LevelDBProto";
option objc_class_prefix = "HLW";

package database;

service Database {
  rpc Get (GetRequest) returns (GetResponse) {}
  rpc Put (PutRequest) returns (PutResponse) {}
  rpc IsMemoryUsageHigh (EmptyParams) returns (MemoryUsageResponse) {}
  rpc KeysToMove (KeysToMoveRequest) returns (KeysToMoveResponse) {}
  rpc Delete(DeleteRequest) returns (DeleteResponse){}
}

message DeleteRequest {
  string key = 1;
}

message DeleteResponse{
  string errormsg = 1;
}



message KeysToMoveRequest{
  string startKey = 1;
  string endKey = 2;

}

message KVpair{
  string key = 1;
  string value = 2;
}

message KeysToMoveResponse{
  repeated KVpair entries = 1;
}

message EmptyParams {
}

message MemoryUsageResponse{
  bool isHigh = 1;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  string value = 2;
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  string errormsg = 1;
}